### ISSUE TITLE
MCQ: MSQ: Give Different ID for generatedOptions

### ISSUE THREAD

**Issue by tshradheya** [TEAMMATES/teammates#8590]
Regression noticed while working on #8584 

Currently, the id for the element used to denote from which choice are options generated is same for MCQ and MSQ.

Since it is `generatedOptions-questionNum` it is different for saved questions. However when new question is created, the NEW_QUESTION=-1 and id is same, hence they misbehave causing either mcq or msq generated from to change without reflecting on the UI.

**Comment by damithc** [comment]
How bad is this bug? Does it hinder typical usage?

**Comment by tshradheya** [comment]
Fortunately it doesn't affect right now because this(duplicate id) only happens when new question is being added and the current code has misplaced brackets leading to undefined `questionNum` https://github.com/TEAMMATES/teammates/issues/8576#issuecomment-370170234

However when I fixed the bracket issue and then if I do something like this:
1. Select Add new MCQ question and choose generate from teams.
2. Cancel question.
3. Again select adding new question. Now even though the UI shows that it will generate from teams, the value is "NONE" in thr HTML because the calling of `toggleMsq...()` made it NONE because for MSQ the checkbox is unchecked.

I have fixed this issue in #8584
